mep/people/admin.py

```diff
@@ -436,8 +471,103 @@ class Media:
         ]
 
 
+class ExportPersonResource(ModelResource):
+    class Meta:
+        model = Person
+        fields = PERSON_IMPORT_EXPORT_COLUMNS
+        export_order = PERSON_IMPORT_EXPORT_COLUMNS
+
+
+class PersonResource(ModelResource):
+    def __init__(self, *x, **y):
+        super().__init__(*x, **y)
+        # list to contain updated objects for batch indexing at end
+        self.objects_to_index = []
+
+    def before_import(self, dataset, *args, **kwargs):
+        # lower and camel_case headers
+        dataset.headers = [x.lower().replace(" ", "_") for x in dataset.headers]
+
+        # turn off indexing temporarily
+        IndexableSignalHandler.disconnect()
+
+        # turn off viaf lookups
+        settings.SKIP_VIAF_LOOKUP = True
+
+    def before_import_row(self, row, **kwargs):
+        """
+        Called on an OrderedDictionary of row attributes.
+        Opportunity to do quick string formatting as a
+        principle of charity to annotators before passing
+        values into django-import-export lookup logic.
+        """
+        # gender to one char
+        gstr = str(row.get("gender")).strip()
+        row["gender"] = gstr[0].upper() if gstr else ""
+
+    def after_save_instance(self, instance, using_transactions, dry_run):
+        """
+        Called when an instance either was or would be saved (depending on dry_run)
+        """
+        self.objects_to_index.append(instance)
+        return super().after_save_instance(instance, using_transactions, dry_run)
+
+    def after_import(self, dataset, result, using_transactions, dry_run, **kwargs):
+        """
+        Called after importing, twice: once with dry_run==True (preview),
+        once dry_run==False. We report how many objects were updated and need to be indexed.
+        We only do so when dry_run is False.
+        """
+        # run parent method
+        super().after_import(dataset, result, using_transactions, dry_run, **kwargs)
+
+        # report how many need indexing
+        logger.debug(
+            f"indexing {len(self.objects_to_index)} objects, dry_run = {dry_run}"
+        )
+
+        # only continue if not a dry run
+        if not dry_run:
+            # re-enable indexing
+            IndexableSignalHandler.connect()
```

Comment:
Could it be this causing 1 or 2 of the failing tests? The signal handler ought to revert to its original condition (connected) but maybe it's not somehow and we need to ensure that

Reply:
ooh good catch, I bet you're right

Reply:
maybe i should re-connect in a `tearDown` method on the testing class. and/or revert to previous value instead of assuming it was previously connected

Reply:
@quadrismegistus quick pr on parasolr to update the disconnect method to return a count of handlers disconnected; could use this to determine whether to reconnect https://github.com/Princeton-CDH/parasolr/pull/84

Reply:
but it is pretty likely a test-only scenario, so your other solution may be simpler + sufficient